modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java

```diff
@@ -1240,6 +1240,79 @@ public void testLimitSearchToPDFFiles() throws Exception {
 			"[Article file 1, Article file 2, PDF file]");
 	}
 
+	@Test
+	public void testLimitSearchToPublishedContents() throws Exception {
+		_updateConfigurationJSON(
+			"queryConfiguration", JSONUtil.put("applyIndexerClauses", false));
+
+		_journalArticles.add(
+			JournalTestUtil.addArticle(
+				_group.getGroupId(), 0,
+				PortalUtil.getClassNameId(JournalArticle.class),
+				HashMapBuilder.put(
+					LocaleUtil.US, "Draft Article"
+				).build(),
+				null,
+				HashMapBuilder.put(
+					LocaleUtil.US, ""
+				).build(),
+				LocaleUtil.getSiteDefault(), true, false, _serviceContext));
+
+		_updateElementInstancesJSON(
+			new Object[] {
+				HashMapBuilder.<String, Object>put(
+					"boost", 1
+				).put(
+					"fields", SXPBlueprintSearchResultTestUtil.FIELDS
+				).put(
+					"fuzziness", "AUTO"
+				).put(
+					"keywords", "${keywords}"
+				).put(
+					"minimum_should_match", 0
+				).put(
+					"operator", "or"
+				).put(
+					"slop", 0
+				).put(
+					"type", "best_fields"
+				).build()
+			},
+			new String[] {"Text Match Over Multiple Fields"});
+
+		_keywords = "Article";
+
+		_assertSearch("[Draft Article]");
+
+		_updateElementInstancesJSON(
+			new Object[] {
+				HashMapBuilder.<String, Object>put(
+					"boost", 1
+				).put(
+					"fields", SXPBlueprintSearchResultTestUtil.FIELDS
+				).put(
+					"fuzziness", "AUTO"
+				).put(
+					"keywords", "${keywords}"
+				).put(
+					"minimum_should_match", 0
+				).put(
+					"operator", "or"
+				).put(
+					"slop", 0
+				).put(
+					"type", "best_fields"
+				).build(),
+				null
+			},
+			new String[] {
+				"Text Match Over Multiple Fields",
+				"Limit Search to Published Contents"
+			});
+
+		_assertSearch("[]");
```

Comment:
Here and a few other places, we want to make sure that the filtering doesn't just remove results but also still allows some through. Add a published JournalArticle to this test that we would still expect to be returned in this last case.